src/end_entity.rs

```diff
@@ -185,4 +193,52 @@ impl<'a> EndEntityCert<'a> {
     pub fn dns_names(&'a self) -> Result<impl Iterator<Item = GeneralDnsNameRef<'a>>, Error> {
         subject_name::list_cert_dns_names(self)
     }
+
+    /// This function tries to verify that the PKCS8 DER encoded `private_key_der` bytes correspond
+    /// to the public key described in the end-entity certificate's subject public key information.
+    /// If the provided key isn't usable for this EE certificate [`Error::CertPrivateKeyMismatch`]
+    /// will be returned.
+    ///
+    /// Private key encodings other than DER PKCS8 are not supported at this time.
```

Comment:
Haven't looked at this in much detail yet, but I feel like the docstring here should be very explicit about what key types are supported, and ideally also make it explicit which key types that rustls supports it doesn't support.

(I feel a little hesitant about merging this without support for all the key types that rustls otherwise supports...)

Reply:
That's fair, I'll set this as a draft until I have a chance to implement the EC1 and raw DER RSA encoding support. I don't think it will be too much work.

Reply:
PKCS1 for RSA was easy. Getting the EC1 support in took some more work. Will call out some points of discussion in some self-review comments.